packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -325,29 +318,128 @@ async function fetchLatestSnapshotCore(
     });
 }
 
-function validateAndEvalBlobsAndTrees(snapshot: IOdspSnapshot) {
-    assert(Array.isArray(snapshot.trees) && snapshot.trees.length > 0,
+interface ISnapshotRequestAndResponseOptions {
+    response: IOdspResponse<ISnapshotContents>,
+    requestUrl: string,
+    requestHeaders: {[index: string]: any},
+}
+
+async function fetchSnapshotContentsCoreV1(
+    odspResolvedUrl: IOdspResolvedUrl,
+    storageToken: string,
+    snapshotOptions: ISnapshotOptions | undefined,
+    controller?: AbortController,
+    epochTracker?: EpochTracker,
+): Promise<ISnapshotRequestAndResponseOptions> {
+    const snapshotUrl = odspResolvedUrl.endpoints.snapshotStorageUrl;
+    const url = `${snapshotUrl}/trees/latest?ump=1`;
+    const formBoundary = uuid();
+    const formParams: string[] = [];
+    formParams.push(`--${formBoundary}`);
+    formParams.push(`Authorization: Bearer ${storageToken}`);
+    formParams.push(`X-HTTP-Method-Override: GET`);
+    if (snapshotOptions !== undefined) {
+        Object.entries(snapshotOptions).forEach(([key, value]) => {
+            if (value !== undefined) {
+                formParams.push(`${key}: ${value}`);
+            }
+        });
+    }
+    if (odspResolvedUrl.sharingLinkToRedeem) {
+        formParams.push(`sl: ${odspResolvedUrl.sharingLinkToRedeem}`);
+    }
+    formParams.push(`_post: 1`);
+    formParams.push(`\r\n--${formBoundary}--`);
+    const postBody = formParams.join("\r\n");
+    const headers: {[index: string]: any} = {
+        "Content-Type": `multipart/form-data;boundary=${formBoundary}`,
+    };
+
+    const fetchOptions = {
+        body: postBody,
+        headers,
+        signal: controller?.signal,
+        method: "POST",
+    };
+    const response = await (epochTracker?.fetchAndParseAsJSON<IOdspSnapshot>(url, fetchOptions, "treesLatest", true) ??
+        fetchAndParseAsJSONHelper<IOdspSnapshot>(url, fetchOptions));
+    const snapshotContents: ISnapshotContents = convertOdspSnapshotToSnapsohtTreeAndBlobs(response.content);
+    const finalSnapshotContents: IOdspResponse<ISnapshotContents> = { ...response, content: snapshotContents };
+    return  {
+        response: finalSnapshotContents,
+        requestHeaders: headers,
+        requestUrl: url,
+    };
+}
+
+async function fetchSnapshotContentsCoreV2(
```

Comment:
same here. Also want to put some warning somewhere that it's experimental functionality.
Would be nice to always log an error to console and telemetry when it's used, such that we can immediately see if someone accidently left localStorage enabled (and maybe hitting problems but not realizing it's because of it)